src/organisms/gv-pagination/gv-pagination.ts

```diff
@@ -0,0 +1,324 @@
+      if (removeL > 0) {
+        addRight = this.center;
+        const diff = this.data.last - this.data.current_page - addRight;
+        if (diff < 0) {
+          removeL += diff;
+        }
+        left = left.slice(removeL);
+      }
+      right = right.slice(0, addRight);
+    }
+    const leftP = left.map(
+      (i) => html`<gv-button
+        ?small="${this.small}"
+        ?medium="${this.medium}"
+        ?large="${this.large}"
+        outlined
+        @gv-button:click="${this.goToPage.bind(this, i)}"
+        >${i}</gv-button
+      >`,
+    );
+    const rightP = right.map(
+      (i) => html`<gv-button
+        ?small="${this.small}"
+        ?medium="${this.medium}"
+        ?large="${this.large}"
+        outlined
+        @gv-button:click="${this.goToPage.bind(this, i)}"
+        >${i}</gv-button
+      >`,
+    );
+
+    leftP.unshift(
+      html`<gv-button
+        ?small="${this.small}"
+        ?medium="${this.medium}"
+        ?large="${this.large}"
+        .disabled="${this.disabled || leftP.length === 0}"
+        .icon="${this.widget ? 'navigation:angle-left' : null}"
+        .title="${i18n('gv-pagination.previous')}"
+        outlined
+        @gv-button:click="${leftP.length === 0 ? () => {} : this.goToPage.bind(this, this.data.current_page - 1)}"
+        >${this.widget ? '' : i18n('gv-pagination.previous')}</gv-button
+      >`,
+    );
+    rightP.push(
+      html`<gv-button
+        ?small="${this.small}"
+        ?medium="${this.medium}"
+        ?large="${this.large}"
+        .disabled="${this.disabled || rightP.length === 0}"
+        .icon="${this.widget ? 'navigation:angle-right' : null}"
+        .title="${i18n('gv-pagination.next')}"
+        outlined
+        @gv-button:click="${rightP.length === 0 ? () => {} : this.goToPage.bind(this, this.data.current_page + 1)}"
+        >${this.widget ? '' : i18n('gv-pagination.next')}</gv-button
+      >`,
+    );
+    return html`${this.widget ? leftP.slice(0, 1) : leftP}
+    ${html`<gv-button
+      .disabled="${this.disabled}"
+      ?small="${this.small}"
+      ?medium="${this.medium}"
+      ?large="${this.large}"
+      primary
+      .title="${this.widget
+        ? i18n('gv-pagination.results', { count: this.data.total })
+        : this.data.current_page + ' / ' + pagination.length}"
+      >${this.widget ? this.data.current_page + ' / ' + pagination.length : this.data.current_page}</gv-button
+    >`}
+    ${this.widget ? rightP.slice(rightP.length - 1, rightP.length) : rightP}`;
+  }
+
+  private get hasData() {
+    return this.data?.total_pages && this.data?.last && this.data?.current_page;
+  }
+
+  private get hide() {
+    return this.hideEmpty && this.data?.total_pages < 2;
+  }
+
+  private get pageSize() {
+    if (this.hasSelect) {
+      return this.data?.size || this.data?.sizes[0];
+    }
+    return null;
+  }
+
+  // @ts-ignore
```

Comment:
still necessary ? one `{ detail }: any` or better type would not be enough ?

Reply:
✅